### PR TITLE
fix: updated all a tags to Link tags in app summary

### LINKED ISSUE #9599: ArgoCD PROJECT UI link is 404 error when using "--rootpath"
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

ArgoCD is deployed using `--rootpath=/argocd`.  ArgoCD overall works fine, however PROJECT link in the UI does not seem to be respecting the `rootpath`.

**To Reproduce**

* Select any ArgoCD application in the web UI.
* Click the root of the application to bring up the Summary screen.
* The link for the `PROJECT` such as `default` does not include the contents of `--rootpath=/argocd`
* Click the link and you get a 404 page not found.

The `Log out` button is also a incorrect in that it protocol switches from HTTPS to HTTP, as I do not have an automatic redirect this results in a 404.  There is no reason that ArgoCD should change protocol upon logout.

**Expected behavior**

The links in the UI should respect the contents of  `--rootpath=/argocd`

**Screenshots**

argocd_url_issue

**Version**

```shell
Not deploying the ArgoCD cli, UI shows: v2.3.4+ac8b7df
```

**Logs**

```
Paste any relevant application logs here.


```

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

Fixes #9599 

This PR updates all anchor tags in app summary to Link tags which automatically handle baseHref/rootpath value.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/applications/components/application-summary/application-summary.tsx [hunk 1]
```diff
@@ -276,9 +277,9 @@ export const ApplicationSummary = (props: {app: models.Application; updateApp: (
                     {urls
                         .map(item => item.split('|'))
                         .map((parts, i) => (
-                            <a key={i} href={parts.length > 1 ? parts[1] : parts[0]} target='__blank'>
+                            <Link key={i} to={parts.length > 1 ? parts[1] : parts[0]} target='__blank'>
```
Comment (alexmt) [inline ui/src/app/applications/components/application-summary/application-summary.tsx:280]: Sorry, I've just realized that links a line 280 and 361 contains absolute URLs (typically external URL) and don't need to be updated. Can you please revert them back to `<a>`

**Review by alexmt — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/application-summary/application-summary.tsx (hunk 1 above)
Comment (gdsoumya) [reply ui/src/app/applications/components/application-summary/application-summary.tsx]: thanks for noticing, reverted those to a tags

**Review by alexmt — CHANGES_REQUESTED** [review]
CI is complaining about linting errors . You can fix it using `yarn lint --project . --fix`

**Review by alexmt — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.7.

**Pull request merged**